python-sdk/tests_integration/databases/test_mysql.py

```diff
@@ -0,0 +1,132 @@
+    [
+        (DEFAULT_CONN_ID, re.compile(r"^(mysql://)(.*)$")),
+        (
+            CUSTOM_CONN_ID,
+            r"^(mysql://)(.*:3306/)(astrodb|tmp_astro)$",
+        ),
+    ],
+    ids=SUPPORTED_CONN_IDS,
+)
+def test_mysql_sqlalchemy_engine(conn_id, expected_uri):
+    """Test getting a mysql based sqla engine."""
+    database = MysqlDatabase(conn_id)
+    engine = database.sqlalchemy_engine
+    assert isinstance(engine, sqlalchemy.engine.base.Engine)
+    url = urlparse(str(engine.url))
+    assert re.match(expected_uri, url.geturl())
+
+
+@pytest.mark.integration
+def test_mysql_run_sql():
+    """Test run_sql against mysql database"""
+    statement = "SELECT 1 + 1;"
+    database = MysqlDatabase(conn_id=CUSTOM_CONN_ID)
+    response = database.run_sql(statement, handler=lambda x: x.first())
+    assert response[0] == 2
+
+
+@pytest.mark.integration
+def test_table_exists_raises_exception():
+    """Test if table exists in mysql database"""
+    database = MysqlDatabase(conn_id=CUSTOM_CONN_ID)
+    table = Table(name="non-existing-table", metadata=Metadata(schema=SCHEMA))
+    assert not database.table_exists(table)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.MYSQL,
+            "table": Table(
+                metadata=Metadata(database=SCHEMA, schema=SCHEMA),
+                columns=[
+                    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                    sqlalchemy.Column("name", sqlalchemy.String(60), nullable=False, key="name"),
+                ],
+            ),
+        }
+    ],
+    indirect=True,
+    ids=["mysql"],
+)
+def test_mysql_create_table_with_columns(database_table_fixture):
+    """Test table creation with columns data"""
+    database, table = database_table_fixture
+
+    statement = f"SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name='{table.name}'"
+    response = database.run_sql(statement, handler=lambda x: x.first())
+    assert response is None
+
+    database.create_table(table)
+    response = database.run_sql(statement, handler=lambda x: x.fetchall())
+    rows = response
+    assert len(rows) == 2
+    assert rows[0][1:4] == (
+        SCHEMA,
+        f"{table.name}",
+        "id",
+    )
+    assert rows[1][1:4] == (
+        SCHEMA,
+        f"{table.name}",
+        "name",
+    )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.MYSQL},
+    ],
+    indirect=True,
+    ids=["mysql"],
+)
+def test_load_pandas_dataframe_to_table(database_table_fixture):
+    """Test load_pandas_dataframe_to_table against mysql"""
+    database, table = database_table_fixture
+
+    pandas_dataframe = pd.DataFrame(data={"id": [1, 2]})
+    database.load_pandas_dataframe_to_table(pandas_dataframe, table)
+
+    statement = f"SELECT * FROM {database.get_table_qualified_name(table)};"
```

Comment:
Here too, we can pass more specific query by passing the column names

Reply:
done